synapseclient/models/table.py

```diff
@@ -489,226 +909,2209 @@ def fill_from_dict(
+                table_id=self.id, synapse_client=synapse_client
+            )
+            for column in column_instances:
+                if column.name not in self.columns:
+                    self.columns[column.name] = column
+
+        if include_activity:
+            self.activity = await Activity.from_parent_async(
+                parent=self, synapse_client=synapse_client
+            )
+
+        self._set_last_persistent_instance()
+        return self
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Table_Delete: {self.name}"
+    )
+    async def delete_async(self, *, synapse_client: Optional[Synapse] = None) -> None:
+        """Delete the table from synapse.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+        Returns:
+            None
+
+        Example: Deleting a table
+            Deleting a table is only supported by the ID of the table.
+
+                import asyncio
+                from synapseclient import Synapse
+
+                syn = Synapse()
+                syn.login()
+
+                async def main():
+                    await Table(id="syn4567").delete_async()
+
+                asyncio.run(main())
+        """
+        if not (self.id or (self.name and self.parent_id)):
+            raise ValueError(
+                "The table must have an id or a " "(name and `parent_id`) set."
+            )
+
+        entity_id = await get_id(entity=self, synapse_client=synapse_client)
+
+        loop = asyncio.get_event_loop()
+        await loop.run_in_executor(
             None,
             lambda: Synapse.get_client(synapse_client=synapse_client).delete(
-                obj=self.id
+                obj=entity_id
             ),
         )
 
-    @classmethod
+    def delete_column(self, name: str) -> None:
+        """
+        Mark a column for deletion. Note that this does not delete the column from
+        Synapse. You must call the `.store()` function on this table class instance to
+        delete the column from Synapse. This is a convenience function to eliminate
+        the need to manually delete the column from the dictionary and add it to the
+        `._columns_to_delete` attribute.
+
+        Arguments:
+            name: The name of the column to delete.
+
+        Returns:
+            None
+
+        Example: Deleting a column
+            This example shows how you may delete a column from a table and then store
+            the change back in Synapse.
+
+                from synapseclient import Synapse
+                from synapseclient.models import Table
+
+                syn = Synapse()
+                syn.login()
+
+                table = Table(
+                    id="syn1234"
+                ).get(include_columns=True)
```

Comment:
I wonder if this would be confusing for our users for `include_columns` default behavior to be false.

Reply:
I agree that it'll make sense to leave this as `True` by default